### ISSUE TITLE
Client Selection on Users Page Impossible in Some Constellations

### ISSUE THREAD

**Issue by tiwalter** [keycloak/keycloak-admin-ui#2508]
### Describe the bug

If I want to assign client roles to a specific user, i run into an issue due to the actual filtering & limitation to 20 clients.

Imagine, you have for example 40 clients with the word "spark" in the client name (client-1-spark, client-2-spark-job, spark, ...).
If you want to select the client "spark", the UI applies a wildcard filter and shows the first 20 clients with "spark" in the name.
The UI does not show the last 20 clients, where the actual "spark" client is part of it.

Here the picture:

![grafik](https://user-images.githubusercontent.com/26794582/165331019-6413ee4e-e46a-45d8-86cd-a6ca117a4f74.png)

This behaviour makes it impossible to select the "spark" client. 

### Version

16.1.0

### Expected behavior

Several behaviours are acceptable:
1. No filter text field and only a drop-down menu with all clients (like in the old Keycloak versions)
2. You can show all clients that match this wildcard filter (no limit to 20)
3. You can paginate the drop-down-menu
4. Allow "exact" matching 

### Actual behavior

Only the first 20 clients that match the filter will be shown in the drop-down menu. No possibility to chose the others in the UI

### How to Reproduce?

- Create >20 clients with e.g. "spark" in the name (client-1-spark, client-2-spark, .... client-25-spark"
- Create a client with the name "spark"
- Try to choose the "spark" client in the "Users" Page -> "Role Mappings" -> "Client Roles"

**Comment by ssilvert** [comment]
Hi @tiwalter.  This is the issues database for the new admin console, which should have exactly what you are looking for.  We will not be updating this behavior in the old console.

Can you give it a try with the latest version of Keycloak and let us know if it meets your needs?

You will need to go to Realm Settings-->Themes and set the admin console theme to "keycloak.v2".

**Comment by tiwalter** [comment]
Hi @ssilvert. Thanks for your answer.

I tested the v18.0.0 with the admin2 preview feature. The core concept behind the new UI will meet the needs. But the functionality is currently buggy ;)

I found some issues in the role assignement:
- The role filtering does not work as expected (switching pages does not have an effect)
- Filter by origin does not work (if you want to choose a specific client or type sth in the search bar, you will get the main search results)

I think the admin v2 theme is still in development?

**Comment by ssilvert** [comment]
Yes, we are still working on it.  The new UI is expected to become the default with Keycloak 19.

**Comment by ssilvert** [comment]
Oh, and your issue should be fixed now by https://github.com/keycloak/keycloak-admin-ui/issues/2375

**Comment by tiwalter** [comment]
Thanks for the infos!

Short recap:
- The described issues will be fixed in the new Keycloak Admin UI
- The new UI will become stable & default with Keycloak 19

I will close this issue now.
